haystack/nodes/prompt/prompt_node.py

```diff
@@ -490,40 +532,41 @@ def invoke(self, *args, **kwargs):
             "best_of": kwargs_with_defaults.get("best_of", 1),
             "logit_bias": kwargs_with_defaults.get("logit_bias", {}),
         }
-        headers = {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
-        response = requests.request(
-            "POST",
-            self.url,
-            headers=headers,
-            data=json.dumps(payload),
-            timeout=float(os.environ.get(HAYSTACK_REMOTE_API_TIMEOUT_SEC, 30)),
-        )
-        res = json.loads(response.text)
+        res = openai_request(url=self.url, api_key=self.api_key, payload=payload)
+        _check_openai_text_completion_answers(result=res, payload=payload)
+        responses = [ans["text"].strip() for ans in res["choices"]]
+        return responses
 
-        if response.status_code != 200:
-            openai_error: OpenAIError
-            if response.status_code == 429:
-                openai_error = OpenAIRateLimitError(f"API rate limit exceeded: {response.text}")
-            else:
-                openai_error = OpenAIError(
-                    f"OpenAI returned an error.\n"
-                    f"Status code: {response.status_code}\n"
-                    f"Response body: {response.text}",
-                    status_code=response.status_code,
-                )
-            raise openai_error
+    def _ensure_token_limit(self, prompt: str) -> str:
+        """Ensure that length of the prompt and answer is within the maximum length of the model.
+        This is accomplished by truncating the prompt text to fit within the max tokens limit of the model.
```

Comment:
Ensure the lenght of the prompt and answer is within the maximum tokens limit of the model. You can shorten the prompt text to fit the limit.